controllers/spiaccesstoken_controller.go

```diff
@@ -322,6 +336,48 @@ func (r *SPIAccessTokenReconciler) oAuthUrlFor(ctx context.Context, at *api.SPIA
 	return oauthBaseUrl + "?state=" + state, nil
 }
 
+func (r *SPIAccessTokenReconciler) refreshToken(ctx context.Context, at *api.SPIAccessToken, sp serviceprovider.ServiceProvider) error {
```

Comment:
I am looking for a better place to put this method. The only suitable place that I can think of is to attach it to the SP factory but that does not seem ideal.
Any suggestions?

Reply:
That is right.